tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsContextPropagator.cs

```diff
@@ -31,7 +36,24 @@ public void Inject<TCarrier>(PathwayContext context, TCarrier headers)
     {
         if (headers is null) { ThrowHelper.ThrowArgumentNullException(nameof(headers)); }
 
-        headers.Add(DataStreamsPropagationHeaders.PropagationKey, PathwayContextEncoder.Encode(context));
+        var encodedBytes = PathwayContextEncoder.Encode(context);
+
+        int base64Length = ((encodedBytes.Length + 2) / 3) * 4;
+        byte[] base64EncodedContextBytes = new byte[base64Length];
+        var status = Base64.EncodeToUtf8(encodedBytes, base64EncodedContextBytes, out _, out int bytesWritten);
+
+        if (status != OperationStatus.Done)
+        {
+            Log.Error("Failed to encode Data Streams context to Base64. OperationStatus: {Status}", status);
+            throw new InvalidOperationException($"Base64 encoding failed with status: {status}");
+        }
+
+        headers.Add(DataStreamsPropagationHeaders.PropagationKeyBase64, base64EncodedContextBytes.AsSpan(0, bytesWritten).ToArray());
```

Comment:
Oops. Now this `ToArray` will allocate a new byte array. I think in most cases `bytesWritten` will be equal to `base64EncodedContextBytes.Length`, so you can use `base64EncodedContextBytes` directly. Maybe add a check in case they are not equal.

Roughly:
```csharp
if (bytesWritten == base64EncodedContextBytes.Length)
{
    headers.Add(DataStreamsPropagationHeaders.PropagationKeyBase64, base64EncodedContextBytes);
}
else
{
    // allocates a new byte array
    headers.Add(DataStreamsPropagationHeaders.PropagationKeyBase64, base64EncodedContextBytes.AsSpan(0, bytesWritten).ToArray());
}
```

Sorry for being so fussy, but we take heap allocations (and their effect on user's apps) very seriously in the tracer 😅

Reply:
Agreed, migrated to your suggestion.